src/CGIHandler.cpp

```diff
@@ -12,10 +10,145 @@ CGIHandler::~CGIHandler()
 {
 }
 
+CGIHandler &CGIHandler::operator=(const CGIHandler &other)
+{
+	if (this != &other)
+	{										// Protect against self-assignment
+		AResponseHandler::operator=(other); // Call the base class assignment operator
+											// Copy or assign other members of CGIHandler if necessary
+	}
+	return *this;
+}
+
 HTTPResponse CGIHandler::handleRequest(const HTTPRequest &request)
 {
-	(void)request;
+	CGIHandler cgiInstance;
+	MetaVariables env;
+	env.HTTPRequestToMetaVars(request, env);
+	std::string cgiOutput = executeCGI(env);
+
+	HTTPResponse response = CGIStringToResponse(cgiOutput);
+
+	// HTTPResponse response;
+	// response.setBody(cgiOutput);
+	// response.setIsCGI(true);
+	// std::cout << response;
+	return response;
+}
+
+char *const *CGIHandler::createArgvForExecve(const MetaVariables &env)
+{
+	std::cout << env;
+	char **argv = new char *[2];
+
+	std::string scriptName = env.getVar("SCRIPT_NAME");
+	std::string pathTranslated = env.getVar("PATH_TRANSLATED");
+	std::string scriptPath = pathTranslated + scriptName;
+
+	if (env.getVar("X_INTERPRETER_PATH") != "")
+	{
+		argv[0] = new char[env.getVar("X_INTERPRETER_PATH").length() + 1];
+		ft_strcpy(argv[0], env.getVar("X_INTERPRETER_PATH").c_str());
+		argv[1] = new char[scriptPath.length() + 1];
+		ft_strcpy(argv[1], scriptPath.c_str());
+		argv[2] = NULL;
+	}
+	else
+	{
+		argv[0] = new char[scriptPath.length() + 1];
+		ft_strcpy(argv[0], scriptPath.c_str());
+		argv[1] = NULL;
+	}
+
+	return argv;
+}
+
+HTTPResponse CGIHandler::CGIStringToResponse(const std::string &cgiOutput)
+{
 	HTTPResponse response;
-	// Determine CGI script path and set up environment
```

Comment:
I ask myself if is conceptually correct to define the HTTPResponse hier. Theoretically the response should be declared on the top of the handleConnection function and I guess it would be good if only one response object would go through the whole process and not to create different response objects in different places. Cause already the HTTPRequest/Parser could "fill" an HTTPResponse with an error code (which would by way bettter than set an error code on HTTPRequest object, which should not have a status code). But this is more something we can think about than a change request.

Reply:
I see what you mean but indeed we have to thing about it, this way also allows us to have a method that returns the response, at the moment it made more sense than to have a method "filling" some of the attribute, but maybe it is not the best.

Actually we should have a red print if we write an error code on top of an already existing error code, I feel this would help a lot for debugging

Reply:
Yes but this is more complex, the functions that call the (various) setErrorCode are called all over the place, so I think it is good to have a safeguard, just before (re)writing an error code into the member variable

Reply:
Do whatever it seems better to you. Maybe write an issue about rewriting status code, which shoudn't happen. Maybe add let the function just return If the statusCode != 0 but I don't know exactly the whole chain.

Reply:
I added it then